src/test/java/io/github/thunderz99/cosmos/CosmosDatabaseTest.java

```diff
@@ -217,11 +218,30 @@ public enum Skill {
     @Test
     public void find_should_work_with_filter() throws Exception {
 
+
+        // aggregate with cross-partition
+        {
+            var join = Join.join("parents", "children");
+
+            var cond = Condition.filter("parents.firstName", "Thomas", "parents.firstName", "Mary Kay", "children.gender", "female", "children.grade <", 6) //
+                    .sort("id", "ASC") //
+                    .limit(10) //
+                    .offset(0);
+
+            var result = db.find(coll, join, cond, "Families").toMap();
+            assertThat(result).hasSize(1);
+            assertThat(result.get(0).get("_partition")).isEqualTo("Families");
+            assertThat(JsonUtil.toListOfMap(JsonUtil.toJson(result.get(0).get("parents"))).stream().anyMatch(item -> item.get("firstName").toString().equals("Thomas"))).isTrue();
+            assertThat(JsonUtil.toListOfMap(JsonUtil.toJson(result.get(0).get("parents"))).stream().anyMatch(item -> item.get("firstName").toString().equals("Mary Kay"))).isTrue();
+            assertThat(JsonUtil.toListOfMap(JsonUtil.toJson(result.get(0).get("children"))).stream().anyMatch(item -> item.get("gender").toString().equals("female"))).isTrue();
+            assertThat(JsonUtil.toListOfMap(JsonUtil.toJson(result.get(0).get("children"))).stream().anyMatch(item -> item.get("grade").toString().equals("5"))).isTrue();
+        }
+
```

Comment:
Please move this test code block, to the end of the method. This test method expects test cases from simple queries  to complex queries.

Or you can create a new test method.